android/src/org/mozilla/firefox/vpn/qt/VPNActivity.java

```diff
@@ -39,6 +42,19 @@ public void onBackPressed() {
     }
   }
 
+  @Override
+  public Object getSystemService (String name){
+
+    if(Build.VERSION.SDK_INT >= 29 && name.equals("clipboard")){
+      // QT will always attempt to read the clipboard if content is there.
+      // since we have no use of the clipboard in android 10+
+      // we _can_  return null 
```

Comment:
This seems a hack. Can you tell me more about why this is needed?

Reply:
I mean ... it is! 
During app startup, but before the mvpn_app startup qt will query the clipboard service, and if that is present it will register a  [ClipBoard ChangedListener](https://developer.android.com/reference/android/content/ClipboardManager.OnPrimaryClipChangedListener) - that will always read the clipboard and the new contents to qt. 
This all happens in the android-qt glue code, so it's unrelated whether we use it on our qt-side or not. 

This is problematic, as android 12 will now notify users on clipboard reads, so every time the app goes foreground with clipboard content present, the warning will show. I think forking/re-writing the glue code is a bit out of scope since qt6 is coming.  So I figured we might just remove the access of the service, which means we would skip that whole initialization flow.
https://code.qt.io/cgit/qt/qtbase.git/tree/src/android/jar/src/org/qtproject/qt5/android/QtNative.java?h=5.9#n594